src/cache.ts

```diff
@@ -53,6 +53,48 @@ const wrapCacheEntry = (entry: CacheEntry): CachePayload => {
   };
 };
 
+/**
+ * Finds the corresponding key in the cache based on the provided key data.
+ * The keys inside the cache are in the format {prefix}::{client_id}::{audience}::{scope}.
+ * The first key in the cache that satisfies the following conditions is returned
+ *  - `prefix` is strict equal to Auth0's internally configured `keyPrefix`
+ *  - `client_id` is strict equal to the `cacheKeyData.client_id`
+ *  - `audience` is strict equal to the `cacheKeyData.audience`
+ *  - `scope` contains at least all the `cacheKeyData.scope` values
+ *  *
+ * @param cacheKeyData The provided cacheKeyData
+ * @param existingCacheKeys A list of existing cache keys
+ */
+const findExistingCacheKey = (
+  cacheKeyData: CacheKeyData,
+  existingCacheKeys: Array<string>
+) => {
+  const { client_id, audience, scope } = cacheKeyData;
+  return existingCacheKeys.find(key => {
+    const [
+      currentPrefix,
+      currentClientId,
+      currentAudience,
+      currentScopes
+    ] = key.split('::');
```

Comment:
I wonder if we're at the point where we could refactor `CacheKeyData` from merely an interface into a class that has these utility methods on it for getting the compound key and also breaking it apart again using `split` like you've done here.

```js
// This could be moved into a CacheKeyData class
const createKey = (e: CacheKeyData) =>
  `${keyPrefix}::${e.client_id}::${e.audience}::${e.scope}`;
```

Also, I would consider renaming `CacheKeyData` to simply `CacheKey`. It's an internal type, so renaming should be fine.

What do you think?

Reply:
I think creating a class is a good idea.
Renaing it to cacheKey could become confusing if we keep current cacheKey variables, which are strings. But I think we can rework that as part of creating the CacheKeyData class

Reply:
Reworked this to use the CacheKey class.